### DISCUSSION TITLE
Backend is offline (network error)

### DISCUSSION THREAD

**Discussion by rtfm87** [amidaware/tacticalrmm#1656] [General]
**Server Info (please complete the following information):**
 - OS: Ubuntu- 22.04
 - Browser: multiple / Safari/Chrome
 -  RMM Version: latest
**Installation Method:**
  - [c ] Standard
 

**Describe the bug**
After a fresh installation on Ubuntu 22.04 i'm not able to login to the webui with rmm.domain.com, there is a "Backend is offline" error. 

The error on browser is "Failed to load resource: net::ERR_SSL_UNRECOGNIZED_NAME_ALERT" at access to the api.
logon to mesh.domain.com works fine.

**Expected behavior**
a fine login in the webui

**Comment by wh1te909** [comment]
that's a very generic error. what troubleshooting steps have you taken? https://docs.tacticalrmm.com/troubleshooting/

**Comment by chrisblunt-codes** [comment]
**Fix for "Backend is Down" / ERR_CERT_COMMON_NAME_INVALID with Wildcard Certificate**

I recently encountered an issue where the Tactical RMM web interface would show "Backend is Down" warnings when accessing rmm.secured.land. In the browser console, I could see the following error:
```
api.rmm.secured.land/_allauth/browser/v1/config/:1 Failed to load resource: net::ERR_CERT_COMMON_NAME_INVALID
```
I ran troubleshooting.sh and found no issues.

**What I checked:**

My wildcard SSL certificate looked valid and had not expired

```Certificate Name: secured.land
Serial Number: 6292737602bd1be7e21bd152b62ed90b43e
Key Type: RSA
Domains: *.secured.land
Expiry Date: 2025-07-23 06:27:22+00:00 (VALID: 89 days)
```

All Tactical RMM services were running properly

1. The Nginx configuration looked correct
2. DNS resolution on the client side showed the correct IP address for api.rmm.secured.land
3. Testing the certificate from the command line showed it was valid:

openssl s_client -connect api.rmm.secured.land:443 -servername api.rmm.secured.land
(This showed a valid certificate with proper verification)
4. Checking the certificate's Subject Alternative Name confirmed it had proper wildcard coverage:
```
sudo openssl x509 -in /etc/letsencrypt/live/secured.land/cert.pem -text -noout | grep -A1 "Subject Alternative Name"
X509v3 Subject Alternative Name: 
    DNS:*.secured.land
```


**The Solution:**
Despite having a valid wildcard certificate covering *.secured.land, the browser consistently rejected it for api.rmm.secured.land with the ERR_CERT_COMMON_NAME_INVALID error.
The fix was to create a dedicated certificate specifically for the API subdomain:

Create a new certificate just for the API subdomain:
```
bashsudo certbot certonly --standalone --pre-hook "systemctl stop nginx" --post-hook "systemctl start nginx" -d api.rmm.secured.land
```

Update the Nginx configuration for the API domain to use this specific certificate:
```
sudo nano /etc/nginx/sites-available/rmm.conf
```

In the api.rmm.secured.land server block, update the certificate paths:
```
server {
    listen 443 ssl reuseport;
    listen [::]:443 ssl;
    server_name api.rmm.secured.land;
    
    # Update these lines to use the new certificate
    ssl_certificate /etc/letsencrypt/live/api.rmm.secured.land/fullchain.pem;
    ssl_certificate_key /etc/letsencrypt/live/api.rmm.secured.land/privkey.pem;
    
    # Rest of the configuration...
}
```

```
Test and restart Nginx:
bashsudo nginx -t
sudo systemctl restart nginx
```


This fixed the issue immediately, and the RMM dashboard now loads without any backend connection errors.

Note:
It seems that there might be some browser-specific handling of wildcard certificates that can cause issues specifically with the API subdomain. Using a dedicated certificate for api.rmm.secured.land bypasses this problem completely. Having said that, I did try multiple PCs and browsers.

Hope this helps someone else who encounters the same issue!

**Comment by Serpher1** [comment]
I'm having a different issue. When I try to access TRMM I get this error. For testing this platform I installed it on VBox on Debian 12. I've reinstalled it 2 times and it's the same. I used "./install.sh --insecure" for a wildcard cert. I can access Mesh and API websites no problem but this one has some trouble.


> Request to resource of another origin blocked: "Same Origin Policy" does not allow loading remote resources from "https://rmmapi.mydomain.com/_allauth/browser/v1/config/" (CORS request failed). Status code: (null).
> 
> Object { stack: "ue@https://192.168.5.10/744c17a1.js:22:21964\ns_</</C.onerror@https://192.168.5.10/744c17a1.js:24:6021\nEventHandlerNonNull*s_</<@htt ps://192.168.5.10/744c17a1.js:24:5998\ns_<@https://192.168.5.10/744c17a1.js:24:5091\nzu@https://192.168.5.10/744c17a1.js:26:512\npromise callback*_request@https://192.168.5.10/744c17a1.js:27:1326\nrequest@https://192.168.5.10/744c17a1.js:26:1979\nIs.prototype[t]@https://192.168.5.10/7 44c17a1.js:27:1787\nah/<@https://192.168.5.10/744c17a1.js:22:16396\nb@htt ps://192.168.5.10/f455f02c.js:1:1318\nsetup/<@https://192.168.5.10/5b95144b .js:1:1082\nXt/</<@https://192.168.5.10/744c17a1.js:14:7459\npo@https://1 0.55.55.6/744c17a1.js:13:1419\nRt@https://192.168.5.10/744c17a1.js:13:149 2\nti/t.__weh@https://192.168.5.10/744c17a1.js:14:7340\ngs@https://10.55. 55.6/744c17a1.js:13:2998\nAf@https://192.168.5.10/744c17a1.js:13:3318\n", message: "Network Error", name: "AxiosError", code: "ERR_NETWORK", config: {…}, request: XMLHttpRequest }

**Reply by Serpher1** [reply]
when I try to log in:

> Object { stack: "ue@https://rmm.mydomain.com/744c17a1.js:22:21964\ns_</</C.onerror@https://rmm.mydomain.com/744c17a1.js:24:6021\nEventHandlerNonNull*s_</<@https://rmm.mydomain.com/744c17a1.js:24:5998\ns_<@https://rmm.mydomain.com/744c17a1.js:24:5091\nzu@https://rmm.mydomain.com/744c17a1.js:26:512\npromise callback*_request@https://rmm.mydomain.com/744c17a1.js:27:1326\nrequest@https://rmm.mydomain.com/744c17a1.js:26:1979\nn/<@https://rmm.mydomain.com/744c17a1.js:27:1938\nah/<@https://rmm.mydomain.com/744c17a1.js:22:16396\ncheckCredentials@https://rmm.mydomain.com/744c17a1.js:35:10955\nw@https://rmm.mydomain.com/744c17a1.js:35:2303\nq@https://rmm.mydomain.com/5b95144b.js:1:739\nIx/n[r]@https://rmm.mydomain.com/744c17a1.js:18:20374\npo@https://rmm.mydomain.com/744c17a1.js:13:1419\nRt@https://rmm.mydomain.com/744c17a1.js:13:1492\nxg@https://rmm.mydomain.com/744c17a1.js:14:32704\nsetup/m/<@https://rmm.mydomain.com/bbb793e6.js:1:1288\npromise callback*m@https://rmm.mydomain.com/bbb793e6.js:1:1243\npo@https://rmm.mydomain.com/744c17a1.js:13:1419\nRt@https://rmm.mydomain.com/744c17a1.js:13:1492\nn@https://rmm.mydomain.com/744c17a1.js:18:8587\nEventListener.handleEvent*Yt@https://rmm.mydomain.com/744c17a1.js:18:7929\nsv@https://rmm.mydomain.com/744c17a1.js:18:8157\nfv@https://rmm.mydomain.com/744c17a1.js:18:9012\nT@https://rmm.mydomain.com/744c17a1.js:14:21675\nx@https://rmm.mydomain.com/744c17a1.js:14:21364\nb@https://rmm.mydomain.com/744c17a1.js:14:20759\nY@https://rmm.mydomain.com/744c17a1.js:14:25179\nrun@https://rmm.mydomain.com/744c17a1.js:9:1808\nq@https://rmm.mydomain.com/744c17a1.js:14:25561\nO@https://rmm.mydomain.com/744c17a1.js:14:24155\nF@https://rmm.mydomain.com/744c17a1.js:14:23945\nb@https://rmm.mydomain.com/744c17a1.js:14:20784\nP@https://rmm.mydomain.com/744c17a1.js:14:22263\nT@https://rmm.mydomain.com/744c17a1.js:14:21544\nx@https://rmm.mydomain.com/744c17a1.js:14:21364\nb@https://rmm.mydomain.com/744c17a1.js:14:20759\nY@https://rmm.mydomain.com/744c17a1.js:14:25179\nrun@https://rmm.mydomain.com/744c17a1.js:9:1808\nq@https://rmm.mydomain.com/744c17a1.js:14:25561\nO@https://rmm.mydomain.com/744c17a1.js:14:24155\nF@https://rmm.mydomain.com/744c17a1.js:14:23945\nb@https://rmm.mydomain.com/744c17a1.js:14:20784\nP@https://rmm.mydomain.com/744c17a1.js:14:22263\nT@https://rmm.mydomain.com/744c17a1.js:14:21544\nx@https://rmm.mydomain.com/744c17a1.js:14:21364\nb@https://rmm.mydomain.com/744c17a1.js:14:20759\nY@https://rmm.mydomain.com/744c17a1.js:14:25179\nrun@https://rmm.mydomain.com/744c17a1.js:9:1808\nq@https://rmm.mydomain.com/744c17a1.js:14:25561\nO@https://rmm.mydomain.com/744c17a1.js:14:24155\n", message: "Network Error", name: "AxiosError", code: "ERR_NETWORK", config: {…}, request: XMLHttpRequest }

**Reply by ralle12345** [reply]
I had the same problem. I was able to login after I opened the api anf mesh urls directly in the browser and added a certificate exception.

**Comment by naciberk** [comment]
Hi guys. i have install the rmm on a new installed ubuntu 22.04 and everything (mesh, api and rmm links) look like fine and works :)
guess what.. same issue with backend on root domain.

my version is 1.3.1
Is there anyone can help me with it.
I am not sure either what is that backend for?
![Screenshot_1](https://github.com/user-attachments/assets/447d495c-c278-4b51-9a56-eae8a46a49f4)

**Comment by logicaladmin** [comment]
It almost feels like a joke. Even after reaching out to support, we’re experiencing this issue on brand new machines with a fresh install, as well as on our old system following our SSL update. You would expect—especially since we pay for support—a straightforward resolution to be easily available for this problem.

**Reply by wh1te909** [reply]
@logicaladmin cut the crap, you're not paying for support. you've opened multiple tickets and we've told you multiple times that support starts at Tier 3 and have even helped you FOR FREE on other support tickets you've opened, even though you are on community support.

**Reply by logicaladmin** [reply]
Your right We don't pay for tier 3 support, but we do pay $88 a month for your software, which is just below competitor pricing. I value the product and your excellent support, but the lack of details on this error remains a real issue. I truly appreciate your past support with general issues, but it’s frustrating to see recurring problems without enough information or resolution, especially for software we’re paying for.

Info
We built a new system with a compatible Ubuntu version. After restoring our backup, we encountered the same error. Rebuilding from scratch without the backup also resulted in the same issue. I'm now considering running it on Debian to determine if that's the problem.

**Reply by logicaladmin** [reply]
Well here's something odd it is now working properly or at least the new build is. I will be testing the restore to the new system as we have old agents on the network we need to get access to
